executor/aggregate_test.go

```diff
@@ -1297,3 +1298,59 @@ func (s *testSuiteAgg) TestIssue20658(c *C) {
 		}
 	}
 }
+
+func (s *testSuiteAgg) TestIssue15284(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t;")
+	tk.MustExec("create table t(a int);")
+	for i := 0; i < 3000; i++ {
+		tk.MustExec(fmt.Sprintf("insert into t values (%d)", i))
+	}
+	tk.MustExec("set @@tidb_hashagg_partial_concurrency=1")
+	tk.MustExec("set @@tidb_hashagg_final_concurrency=1")
+
+	rows := tk.MustQuery("SELECT distinct a FROM t limit 3").Rows()
+	c.Assert(len(rows), Equals, 3)
+	assertNotExistSameElement(c, rows)
+	assertTableReaderActRows(c, rows, "1024")
+
+	rows = tk.MustQuery("SELECT distinct a FROM t limit 2000").Rows()
+	c.Assert(len(rows), Equals, 2000)
+	assertNotExistSameElement(c, rows)
+	assertTableReaderActRows(c, rows, "2048")
+
+	tk.MustExec("drop table if exists t;")
+	tk.MustExec("CREATE TABLE t (a int, b int)")
+	tk.MustExec("insert into t values (0, 0), (1, 1), (1, 2), (1, 3), (2, 4), (2, 5), (2, 6), (3, 7), (3, 10), (3, 11), (12, 12), (12, 13), (14, 14), (14, 15), (20, 20), (20, 21), (20, 22), (23, 23), (23, 24), (23, 25), (31, 30), (31, 31), (31, 32), (33, 33), (33, 34), (33, 35), (36, 36), (80, 80), (90, 90), (100, 100)")
+	rows = tk.MustQuery("SELECT distinct a, b FROM t limit 20").Rows()
+	c.Assert(len(rows), Equals, 20)
+	assertNotExistSameElement(c, rows)
+	assertTableReaderActRows(c, rows, "1024")
+}
+
+func assertNotExistSameElement(c *C, rows [][]interface{}) {
+	valSet := set.NewStringSet()
+	for _, row := range rows {
+		var key string
+		for _, r := range row {
+			if data, ok := r.(string); ok {
+				key += data
+			}
+		}
+		c.Assert(valSet.Exist(key), Equals, false)
+		valSet.Insert(key)
+	}
+}
+
+func assertTableReaderActRows(c *C, rows [][]interface{}, actRows string) {
+	for _, row := range rows {
+		if s, ok := row[0].(string); ok {
+			if strings.Contains(s, "TableReader") {
```

Comment:
I think there is no string contains "TableReader", because the rows is the result for the SQL, not EXPLAIN ANALYZE